src/code/code_800F9280.c

```diff
@@ -369,292 +407,373 @@ void Audio_ProcessSeqCmd(u32 cmd) {
             }
 
-            if (gAudioContext.seqPlayers[playerIdx].enabled) {
+            // Only process setup commands if `seqPlayerIndex` if no longer playing
+            // i.e. the `seqPlayer` is no longer enabled
+            if (gAudioContext.seqPlayers[seqPlayerIndex].enabled) {
                 continue;
             }
 
-            for (j = 0; j < D_8016E750[playerIdx].unk_4D; j++) {
-                temp_a0 = (D_8016E750[playerIdx].unk_2C[j] & 0x00F00000) >> 20;
-                temp_s1 = (D_8016E750[playerIdx].unk_2C[j] & 0x000F0000) >> 16;
-                temp_s0_3 = (D_8016E750[playerIdx].unk_2C[j] & 0xFF00) >> 8;
-                temp_a3_3 = D_8016E750[playerIdx].unk_2C[j] & 0xFF;
+            for (j = 0; j < gActiveSeqs[seqPlayerIndex].setupCmdNum; j++) {
+                setupOp = (gActiveSeqs[seqPlayerIndex].setupCmd[j] & 0x00F00000) >> 20;
+                targetSeqPlayerIndex = (gActiveSeqs[seqPlayerIndex].setupCmd[j] & 0x000F0000) >> 16;
+                setupVal2 = (gActiveSeqs[seqPlayerIndex].setupCmd[j] & 0xFF00) >> 8;
+                setupVal1 = gActiveSeqs[seqPlayerIndex].setupCmd[j] & 0xFF;
 
-                switch (temp_a0) {
-                    case 0:
-                        Audio_SetVolScale(temp_s1, 1, 0x7F, temp_a3_3);
+                switch (setupOp) {
+                    case SEQCMD_SUB_OP_SETUP_RESTORE_VOLUME:
+                        // Restore `targetSeqPlayerIndex` volume back to normal levels
+                        Audio_SetVolumeScale(targetSeqPlayerIndex, VOL_SCALE_INDEX_FANFARE, 0x7F, setupVal1);
                         break;
-                    case 7:
-                        if (D_8016E348[playerIdx] == temp_a3_3) {
-                            Audio_SetVolScale(temp_s1, 1, 0x7F, temp_s0_3);
+
+                    case SEQCMD_SUB_OP_SETUP_RESTORE_VOLUME_IF_QUEUED:
+                        // Restore `targetSeqPlayerIndex` volume back to normal levels,
+                        // but only if the number of sequence queue requests from `sSeqRequests`
+                        // exactly matches the argument to the command
+                        if (setupVal1 == sNumSeqRequests[seqPlayerIndex]) {
+                            Audio_SetVolumeScale(targetSeqPlayerIndex, VOL_SCALE_INDEX_FANFARE, 0x7F, setupVal2);
                         }
                         break;
-                    case 1:
-                        Audio_SeqCmd3(playerIdx, D_8016E750[playerIdx].unk_254);
+
+                    case SEQCMD_SUB_OP_SETUP_SEQ_UNQUEUE:
+                        // Unqueue `seqPlayerIndex` from sSeqRequests
+                        // Note: this command does not work as intended as unqueueing
+                        // the seqeuence relies on `gActiveSeqs[seqPlayerIndex].seqId`
+                        // However, `gActiveSeqs[seqPlayerIndex].seqId` is reset before the sequence on `seqPlayerIndex`
+                        // is no longer playing, i.e. before this command can run.
```

Comment:
Exactly, when `SEQCMD_STOP_SEQUENCE` -> `Audio_StopSequenceNow` is called, `gActiveSeqs[seqPlayerIndex].seqId = NA_BGM_DISABLED;` is applied but `prevSeqId` remains untouched. 

```
void Audio_StopSequenceNow(u8 seqPlayerIndex, u16 fadeTimer) {
    Audio_QueueCmdS32(0x83000000 | ((u8)seqPlayerIndex << 16),
                      (fadeTimer * (u16)gAudioContext.audioBufferParameters.updatesPerFrame) / 4);
    gActiveSeqs[seqPlayerIndex].seqId = NA_BGM_DISABLED;
}
```

But this is applied when `Audio_QueueCmdS32` is requested, so it will take time for the audio thread to pick up this command, disable the seqPlayer, and set `gAudioContext.seqPlayers[seqPlayerIndex].enabled` to false, which will trigger the setup commands. But then it uses `gActiveSeqs[seqPlayerIndex].seqId` expecting it will have the seqId needed to unqueue, but it's gone. 

`gActiveSeqs[seqPlayerIndex].prevSeqId` remains untouched in `Audio_StopSequenceNow` though so using that instead would have worked

Reply:
so this may be a bit verbose but to complete,
```suggestion
                        // However, `gActiveSeqs[seqPlayerIndex].seqId` is reset when the sequence on `seqPlayerIndex`
                        // is requested to stop, i.e. before the sequence is disabled and setup commands (including this command) can run.
```
since, though it's not worth mentioning explicitly, there's also possibly `fadeTimer` as a delay in the way and not just the processing delays